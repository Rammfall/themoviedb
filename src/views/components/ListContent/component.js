import React from 'react'
import { Col, Row, Typography } from 'antd'
import PropTypes from 'prop-types'

import PrivateLayout from 'Views/layouts/PrivateLayout'
import RenderCtrl from 'Views/components/RenderCtrl'
import Loader from 'Views/components/Loader'
import EmptyState from 'Views/components/EmptyState'

const ListContent = ({
  isEmpty,
  isLoading,
  title,
  titleEmptyState,
  list
}) => (
  <PrivateLayout>
    <Row>
      <Col
        offset={2}
        span={20}
      >
        <div className="top-margin">
          <Typography.Title>{title}</Typography.Title>
        </div>
      </Col>
    </Row>
    <RenderCtrl
      isEmpty={isEmpty}
      isLoading={isLoading}
      renderList={list}
      renderEmpty={<EmptyState title={titleEmptyState} />}
      renderLoading={<Loader />}
    />
  </PrivateLayout>
)

ListContent.propTypes = {
  isEmpty: PropTypes.bool.isRequired,
  isLoading: PropTypes.bool,
  title: PropTypes.string.isRequired,
  titleEmptyState: PropTypes.string.isRequired,
  list: PropTypes.element.isRequired
}

ListContent.defaultProps = {
  isLoading: true
}

export default ListContent
