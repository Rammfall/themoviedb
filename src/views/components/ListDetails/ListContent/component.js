import React from 'react'
import { Col, Row, Typography } from 'antd'
import { MinusCircleOutlined } from '@ant-design/icons'
import PropTypes from 'prop-types'

import RenderCtrl from 'Views/components/RenderCtrl'
import Loader from 'Views/components/Loader'
import EmptyState from 'Views/components/EmptyState'
import ListMovies from './ListMovies'

const ListContentComponent = ({
  title,
  isEmpty,
  isEmptyTitle
}) => (
  <>
    <Row>
      <Col
        offset={2}
        span={20}
      >
        <div className='top-margin'>
          <Typography.Title>
            {title}
            {' '}
            <MinusCircleOutlined />
          </Typography.Title>
        </div>
      </Col>
    </Row>
    <RenderCtrl
      isEmpty={isEmpty}
      isLoading={false}
      renderLoading={<Loader />}
      renderEmpty={<EmptyState title={isEmptyTitle} />}
      renderList={<ListMovies />}
    />
  </>
)

ListContentComponent.propTypes = {
  title: PropTypes.string.isRequired,
  isEmptyTitle: PropTypes.string.isRequired,
  isEmpty: PropTypes.bool.isRequired
}

export default ListContentComponent
