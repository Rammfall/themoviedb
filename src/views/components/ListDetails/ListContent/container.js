import React from 'react'
import { compose } from 'ramda'
import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'
import { injectIntl } from 'react-intl'
import PropTypes from 'prop-types'

import { getListDetailsMoviesSelector, getListTitleSelector } from 'Store/concepts/listDetails/selectors'

import ListContentComponent from './component'

const ListContent = ({
  title,
  isEmpty,
  intl: { formatMessage }
}) => (
  <ListContentComponent
    isEmptyTitle={formatMessage({ id: 'movies.empty' })}
    isEmpty={isEmpty}
    title={title}
  />
)

ListContent.propTypes = {
  title: PropTypes.string.isRequired,
  intl: PropTypes.shape({
    formatMessage: PropTypes.func.isRequired
  }).isRequired,
  isEmpty: PropTypes.bool.isRequired
}

const mapStateToProps = (state, { match: { params: { id } } }) => ({
  title: getListTitleSelector(state, id),
  isEmpty: getListDetailsMoviesSelector(state, id).isEmpty
})

export { ListContent as ListContentContainer }
export default compose(
  withRouter,
  connect(mapStateToProps),
  injectIntl
)(ListContent)
