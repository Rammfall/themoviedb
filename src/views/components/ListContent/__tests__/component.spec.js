import React from 'react'
import { shallow } from 'enzyme'

import ListContent from '../component'

describe('ListContent', () => {
  const Component = () => <div>Test</div>
  const props = {
    isEmpty: false,
    title: 'test',
    titleEmptyState: 'test',
    list: <Component />
  }

  it('matches snapshot', () => {
    const component = shallow(
      <ListContent {...props} />
    )

    expect(component).toMatchSnapshot()
  })
})
