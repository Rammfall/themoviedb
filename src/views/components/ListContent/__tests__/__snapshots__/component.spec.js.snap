// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListContent matches snapshot 1`] = `
<PrivateLayout>
  <Row>
    <Col
      offset={2}
      span={20}
    >
      <div
        className="top-margin"
      >
        <Title>
          test
        </Title>
      </div>
    </Col>
  </Row>
  <RenderCtrl
    isEmpty={false}
    isLoading={true}
    renderEmpty={
      <EmptyState
        title="test"
      />
    }
    renderList={<Component />}
    renderLoading={<Loader />}
  />
</PrivateLayout>
`;
